bfetch (2)

streamich opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/44326
Partially addresses: https://github.com/elastic/kibana/issues/44393

- Moves `batched_fetch` to the NP, now it is `plugins.bfetch.batchedFunction`.
- Introduces `plugins.bfetch.addBatchProcessingRoute` on server-side which works with client-side `plugins.bfetch.batchedFunction`.
- Moves legacy interpreter server-side function execution to NP `expressions` plugin.
- Improves README for `bfetch` plugin.
- Adds `bfetch_explorer` example plugin.

To see the `bfetch_explorer` example plugin run

```shell
yarn start --run-examples
```

![image](https://user-images.githubusercontent.com/9773803/71888566-bb675200-3140-11ea-9ade-a35bd302bcae.png)


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [~~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
- ~~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- ~~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [x] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

---

# Dev Docs

Request batching and response streaming functionality of legacy Interpreter plugin has been moved out into a separate `bfetch` Kibana platform plugin. Now every plugin can create server endpoints and browser wrappers that can batch HTTP requests and stream responses back.

As an example, we will create a batch processing endpoint that receives a number then doubles it
and streams it back. We will also consider the number to be time in milliseconds
and before streaming the number back the server will wait for the specified number of
milliseconds.

To do that, first create server-side batch processing route using `addBatchProcessingRoute`.

```ts
plugins.bfetch.addBatchProcessingRoute<{ num: number }, { num: number }>(
  '/my-plugin/double',
  () => ({
    onBatchItem: async ({ num }) => {
      // Validate inputs.
      if (num < 0) throw new Error('Invalid number');
      // Wait number of specified milliseconds.
      await new Promise(r => setTimeout(r, num));
      // Double the number and send it back.
      return { num: 2 * num };
    },
  })
);
```

Now on client-side create `double` function using `batchedFunction`.
The newly created `double` function can be called many times and it
will package individual calls into batches and send them to the server.

```ts
const double = plugins.bfetch.batchedFunction<{ num: number }, { num: number }>({
  url: '/my-plugin/double',
});
```

Note: the created `double` must accept a single object argument (`{ num: number }` in this case)
and it will return a promise that resolves into an object, too (also `{ num: number }` in this case).

Use the `double` function.

```ts
double({ num: 1 }).then(console.log, console.error); // { num: 2 }
double({ num: 2 }).then(console.log, console.error); // { num: 4 }
double({ num: 3 }).then(console.log, console.error); // { num: 6 }
```

elasticmachine commented:
Pinging @elastic/kibana-app-arch (Team:AppArch)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/)
* Commit: 37c88808dfca9746cd26a51e3e7ca1222654daf3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/","number":"16964","commit":"37c88808dfca9746cd26a51e3e7ca1222654daf3"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/)
* Commit: d9dfdd19004552a505baa3387038ae9def420bfd

### History
* :broken_heart: [Build #16964](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/) failed 37c88808dfca9746cd26a51e3e7ca1222654daf3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/","number":"16964","commit":"37c88808dfca9746cd26a51e3e7ca1222654daf3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/","number":"17406","commit":"d9dfdd19004552a505baa3387038ae9def420bfd"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/)
* Commit: 5b66eb545874d3640332b20293bdd4f2af0e0007

### History
* :broken_heart: [Build #17406](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/) failed d9dfdd19004552a505baa3387038ae9def420bfd
* :broken_heart: [Build #16964](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/) failed 37c88808dfca9746cd26a51e3e7ca1222654daf3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/","number":"16964","commit":"37c88808dfca9746cd26a51e3e7ca1222654daf3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/","number":"17406","commit":"d9dfdd19004552a505baa3387038ae9def420bfd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/","number":"17720","commit":"5b66eb545874d3640332b20293bdd4f2af0e0007"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/)
* Commit: 4aff413fdcb0cc557830003c7bd534db7aecc875

### History
* :broken_heart: [Build #17720](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/) failed 5b66eb545874d3640332b20293bdd4f2af0e0007
* :broken_heart: [Build #17406](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/) failed d9dfdd19004552a505baa3387038ae9def420bfd
* :broken_heart: [Build #16964](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/) failed 37c88808dfca9746cd26a51e3e7ca1222654daf3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/","number":"16964","commit":"37c88808dfca9746cd26a51e3e7ca1222654daf3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/","number":"17406","commit":"d9dfdd19004552a505baa3387038ae9def420bfd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/","number":"17720","commit":"5b66eb545874d3640332b20293bdd4f2af0e0007"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/","number":"17733","commit":"4aff413fdcb0cc557830003c7bd534db7aecc875"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/)
* Commit: 4b543f72d8fd68306a2799f485b3263a697ef57c

### History
* :broken_heart: [Build #17733](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/) failed 4aff413fdcb0cc557830003c7bd534db7aecc875
* :broken_heart: [Build #17720](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/) failed 5b66eb545874d3640332b20293bdd4f2af0e0007
* :broken_heart: [Build #17406](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/) failed d9dfdd19004552a505baa3387038ae9def420bfd
* :broken_heart: [Build #16964](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/) failed 37c88808dfca9746cd26a51e3e7ca1222654daf3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/","number":"16964","commit":"37c88808dfca9746cd26a51e3e7ca1222654daf3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/","number":"17406","commit":"d9dfdd19004552a505baa3387038ae9def420bfd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/","number":"17720","commit":"5b66eb545874d3640332b20293bdd4f2af0e0007"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/","number":"17733","commit":"4aff413fdcb0cc557830003c7bd534db7aecc875"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/","number":"17739","commit":"4b543f72d8fd68306a2799f485b3263a697ef57c"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/)
* Commit: ba85c4a7cda7d97eb0370dd44f35657188bbc580

### History
* :broken_heart: [Build #17739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/) failed 4b543f72d8fd68306a2799f485b3263a697ef57c
* :broken_heart: [Build #17733](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/) failed 4aff413fdcb0cc557830003c7bd534db7aecc875
* :broken_heart: [Build #17720](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/) failed 5b66eb545874d3640332b20293bdd4f2af0e0007
* :broken_heart: [Build #17406](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/) failed d9dfdd19004552a505baa3387038ae9def420bfd
* :broken_heart: [Build #16964](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/) failed 37c88808dfca9746cd26a51e3e7ca1222654daf3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16964/","number":"16964","commit":"37c88808dfca9746cd26a51e3e7ca1222654daf3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/","number":"17406","commit":"d9dfdd19004552a505baa3387038ae9def420bfd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/","number":"17720","commit":"5b66eb545874d3640332b20293bdd4f2af0e0007"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/","number":"17733","commit":"4aff413fdcb0cc557830003c7bd534db7aecc875"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/","number":"17739","commit":"4b543f72d8fd68306a2799f485b3263a697ef57c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/","number":"17760","commit":"ba85c4a7cda7d97eb0370dd44f35657188bbc580"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/)
* Commit: 7195d750c225d488db52c3baf80f83c6ddb49829

### History
* :broken_heart: [Build #17760](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/) failed ba85c4a7cda7d97eb0370dd44f35657188bbc580
* :broken_heart: [Build #17739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/) failed 4b543f72d8fd68306a2799f485b3263a697ef57c
* :broken_heart: [Build #17733](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/) failed 4aff413fdcb0cc557830003c7bd534db7aecc875
* :broken_heart: [Build #17720](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/) failed 5b66eb545874d3640332b20293bdd4f2af0e0007
* :broken_heart: [Build #17406](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/) failed d9dfdd19004552a505baa3387038ae9def420bfd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17406/","number":"17406","commit":"d9dfdd19004552a505baa3387038ae9def420bfd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/","number":"17720","commit":"5b66eb545874d3640332b20293bdd4f2af0e0007"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/","number":"17733","commit":"4aff413fdcb0cc557830003c7bd534db7aecc875"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/","number":"17739","commit":"4b543f72d8fd68306a2799f485b3263a697ef57c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/","number":"17760","commit":"ba85c4a7cda7d97eb0370dd44f35657188bbc580"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/","number":"18001","commit":"7195d750c225d488db52c3baf80f83c6ddb49829"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/)
* Commit: ce0133480fb91beb258858eb6622c1feeb056de8

### History
* :green_heart: [Build #18001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/) succeeded 7195d750c225d488db52c3baf80f83c6ddb49829
* :broken_heart: [Build #17760](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/) failed ba85c4a7cda7d97eb0370dd44f35657188bbc580
* :broken_heart: [Build #17739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/) failed 4b543f72d8fd68306a2799f485b3263a697ef57c
* :broken_heart: [Build #17733](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/) failed 4aff413fdcb0cc557830003c7bd534db7aecc875
* :broken_heart: [Build #17720](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/) failed 5b66eb545874d3640332b20293bdd4f2af0e0007

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17720/","number":"17720","commit":"5b66eb545874d3640332b20293bdd4f2af0e0007"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/","number":"17733","commit":"4aff413fdcb0cc557830003c7bd534db7aecc875"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/","number":"17739","commit":"4b543f72d8fd68306a2799f485b3263a697ef57c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/","number":"17760","commit":"ba85c4a7cda7d97eb0370dd44f35657188bbc580"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/","number":"18001","commit":"7195d750c225d488db52c3baf80f83c6ddb49829"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/","number":"18062","commit":"ce0133480fb91beb258858eb6622c1feeb056de8"}]}
PIPELINE-->

streamich commented on src/plugins/expressions/server/legacy.ts:
```diff
@@ -0,0 +1,140 @@
+  /**
+   * Register the endpoint that executes a batch of functions, and sends the
+   * result back as a single response.
+   */
```
Improve this comment.

streamich commented on examples/README.md:
````diff
@@ -6,3 +6,5 @@ This folder contains example plugins.  To run the plugins in this folder, use th
 ```
 
+Plugins in this folder are not auto-discovered, to make Kibana load your plugin
+from this folder add it to `src/cli/serve/serve.js`.
````
This is not the case anymore.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/)
* Commit: 48af298f9c858d5838c8fbf24ef6b8c88216e28a

### History
* :green_heart: [Build #18062](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/) succeeded ce0133480fb91beb258858eb6622c1feeb056de8
* :green_heart: [Build #18001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/) succeeded 7195d750c225d488db52c3baf80f83c6ddb49829
* :broken_heart: [Build #17760](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/) failed ba85c4a7cda7d97eb0370dd44f35657188bbc580
* :broken_heart: [Build #17739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/) failed 4b543f72d8fd68306a2799f485b3263a697ef57c
* :broken_heart: [Build #17733](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/) failed 4aff413fdcb0cc557830003c7bd534db7aecc875

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17733/","number":"17733","commit":"4aff413fdcb0cc557830003c7bd534db7aecc875"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/","number":"17739","commit":"4b543f72d8fd68306a2799f485b3263a697ef57c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/","number":"17760","commit":"ba85c4a7cda7d97eb0370dd44f35657188bbc580"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/","number":"18001","commit":"7195d750c225d488db52c3baf80f83c6ddb49829"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/","number":"18062","commit":"ce0133480fb91beb258858eb6622c1feeb056de8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/","number":"18356","commit":"48af298f9c858d5838c8fbf24ef6b8c88216e28a"}]}
PIPELINE-->

streamich commented on src/plugins/bfetch/README.md:
```diff
@@ -3,7 +3,54 @@
+and before streaming the number back the server will wait for the specified number of
+milliseconds.
+
+To do that, first create server-side batch processing route using [`addBatchProcessingRoute](./docs/server/reference.md#addBatchProcessingRoute).
```
```suggestion
To do that, first create server-side batch processing route using [`addBatchProcessingRoute`](./docs/server/reference.md#addBatchProcessingRoute).
```

streamich commented on src/plugins/bfetch/README.md:
````diff
@@ -3,7 +3,54 @@
+);
+```
+
+Now on client-side create `double` function using [`batchedFunction](./docs/browser/reference.md#batchedFunction).
````
```suggestion
Now on client-side create `double` function using [`batchedFunction`](./docs/browser/reference.md#batchedFunction).
```

streamich commented on examples/bfetch_explorer/public/containers/app/sidebar/index.tsx:
```diff
@@ -0,0 +1,56 @@
+interface SidebarProps {}
+
+export const Sidebar: React.FC<SidebarProps> = () => {
+  const {} = useDeps();
```
```suggestion
```

streamich commented on examples/bfetch_explorer/public/components/count_until/index.tsx:
```diff
@@ -0,0 +1,93 @@
+
+  if (showingResults) {
+    return (
+      <EuiForm data-test-subj="DoubleIntegers">
```
```suggestion
      <EuiForm data-test-subj="CountUntil">
```

streamich commented on examples/bfetch_explorer/public/components/count_until/index.tsx:
```diff
@@ -0,0 +1,93 @@
+  }
+
+  return (
+    <EuiForm data-test-subj="DoubleIntegers">
```
```suggestion
    <EuiForm data-test-subj="CountUntil">
```

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/)
* Commit: 627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8

### History
* :green_heart: [Build #18356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/) succeeded 48af298f9c858d5838c8fbf24ef6b8c88216e28a
* :green_heart: [Build #18062](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/) succeeded ce0133480fb91beb258858eb6622c1feeb056de8
* :green_heart: [Build #18001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/) succeeded 7195d750c225d488db52c3baf80f83c6ddb49829
* :broken_heart: [Build #17760](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/) failed ba85c4a7cda7d97eb0370dd44f35657188bbc580
* :broken_heart: [Build #17739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/) failed 4b543f72d8fd68306a2799f485b3263a697ef57c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17739/","number":"17739","commit":"4b543f72d8fd68306a2799f485b3263a697ef57c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/","number":"17760","commit":"ba85c4a7cda7d97eb0370dd44f35657188bbc580"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/","number":"18001","commit":"7195d750c225d488db52c3baf80f83c6ddb49829"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/","number":"18062","commit":"ce0133480fb91beb258858eb6622c1feeb056de8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/","number":"18356","commit":"48af298f9c858d5838c8fbf24ef6b8c88216e28a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/","number":"18676","commit":"627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/)
* Commit: 148c9362d2f5fa8c2a3da4e564972e9c033bf5ab

### History
* :broken_heart: [Build #18676](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/) failed 627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8
* :green_heart: [Build #18356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/) succeeded 48af298f9c858d5838c8fbf24ef6b8c88216e28a
* :green_heart: [Build #18062](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/) succeeded ce0133480fb91beb258858eb6622c1feeb056de8
* :green_heart: [Build #18001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/) succeeded 7195d750c225d488db52c3baf80f83c6ddb49829
* :broken_heart: [Build #17760](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/) failed ba85c4a7cda7d97eb0370dd44f35657188bbc580

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17760/","number":"17760","commit":"ba85c4a7cda7d97eb0370dd44f35657188bbc580"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/","number":"18001","commit":"7195d750c225d488db52c3baf80f83c6ddb49829"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/","number":"18062","commit":"ce0133480fb91beb258858eb6622c1feeb056de8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/","number":"18356","commit":"48af298f9c858d5838c8fbf24ef6b8c88216e28a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/","number":"18676","commit":"627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/","number":"18928","commit":"148c9362d2f5fa8c2a3da4e564972e9c033bf5ab"}]}
PIPELINE-->

lukasolson commented on src/plugins/bfetch/common/buffer/tests/timed_item_buffer.test.ts:
```diff
@@ -0,0 +1,105 @@
+import { ItemBuffer } from '../item_buffer';
+
+describe('TimedItemBuffer', () => {
+  runItemBufferTests((TimedItemBuffer as unknown) as new () => ItemBuffer<any>);
```
This seems odd to me... Is there any way to avoid all of these `as` casts? I get that we're just running the same tests with `TimedItemBuffer` but this just doesn't sit right.

streamich replied:
Yes, I made `maxItemAge` optional in `TimeItemBuffer` and now we don't need these casts.

lukasolson commented on src/plugins/kibana_utils/common/of.ts:
```diff
@@ -0,0 +1,37 @@
+  try {
+    return [await promise, undefined, true];
+  } catch (error) {
+    return [, error, false];
```
Not a big deal but this should probably follow the same behavior as above:

```suggestion
    return [undefined, error, false];
```

as they are actually a bit different in implementation (although I'm not sure it would make a difference functionally):

![image](https://user-images.githubusercontent.com/1178348/72110283-0924bb80-32f5-11ea-8f63-c1c542d60869.png)

lukasolson replied:
Where is this actually used? I don't see any usages of it.

streamich replied:
It is used in tests, here:

- `bfetch/public/batching/create_streaming_batched_function.test.ts`

It is very useful in tests, to get the error of a promise or make sure you don't have and `UNHANDLED REJECTION` error.

lukasolson commented:
Also, I know you had mentioned adding in functional tests from the example plugin, is that something you're wanting to tackle in this PR or in a future one?

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/)
* Commit: 2f77faafa1f3571c90b0f08572bc8289e93e708d

### History
* :green_heart: [Build #18928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/) succeeded 148c9362d2f5fa8c2a3da4e564972e9c033bf5ab
* :broken_heart: [Build #18676](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/) failed 627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8
* :green_heart: [Build #18356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/) succeeded 48af298f9c858d5838c8fbf24ef6b8c88216e28a
* :green_heart: [Build #18062](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/) succeeded ce0133480fb91beb258858eb6622c1feeb056de8
* :green_heart: [Build #18001](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/) succeeded 7195d750c225d488db52c3baf80f83c6ddb49829

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18001/","number":"18001","commit":"7195d750c225d488db52c3baf80f83c6ddb49829"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/","number":"18062","commit":"ce0133480fb91beb258858eb6622c1feeb056de8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/","number":"18356","commit":"48af298f9c858d5838c8fbf24ef6b8c88216e28a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/","number":"18676","commit":"627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/","number":"18928","commit":"148c9362d2f5fa8c2a3da4e564972e9c033bf5ab"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/","number":"19152","commit":"2f77faafa1f3571c90b0f08572bc8289e93e708d"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/)
* Commit: eaf01bbfa0626db1bc546c83c2b33f35faeb2f95

### History
* :broken_heart: [Build #19152](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/) failed 2f77faafa1f3571c90b0f08572bc8289e93e708d
* :green_heart: [Build #18928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/) succeeded 148c9362d2f5fa8c2a3da4e564972e9c033bf5ab
* :broken_heart: [Build #18676](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/) failed 627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8
* :green_heart: [Build #18356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/) succeeded 48af298f9c858d5838c8fbf24ef6b8c88216e28a
* :green_heart: [Build #18062](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/) succeeded ce0133480fb91beb258858eb6622c1feeb056de8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18062/","number":"18062","commit":"ce0133480fb91beb258858eb6622c1feeb056de8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/","number":"18356","commit":"48af298f9c858d5838c8fbf24ef6b8c88216e28a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/","number":"18676","commit":"627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/","number":"18928","commit":"148c9362d2f5fa8c2a3da4e564972e9c033bf5ab"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/","number":"19152","commit":"2f77faafa1f3571c90b0f08572bc8289e93e708d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/","number":"19154","commit":"eaf01bbfa0626db1bc546c83c2b33f35faeb2f95"}]}
PIPELINE-->

streamich commented on src/plugins/bfetch/common/buffer/create_batched_function.ts:
```diff
@@ -0,0 +1,49 @@
+  onCall: (...args: Parameters<Func>) => [ReturnType<Func>, BatchEntry];
+  onBatch: (items: BatchEntry[]) => void;
+  flushOnMaxItems?: ItemBufferParams<any>['flushOnMaxItems'];
+  maxItemAge?: TimedItemBufferParams<any>['maxItemAge'];
```
Done.

streamich commented on examples/bfetch_explorer/public/components/count_until/index.tsx:
```diff
@@ -0,0 +1,93 @@
+
+  return (
+    <EuiForm data-test-subj="CountUntil">
+      <EuiFormRow label="Numbers in ms separated by new line" fullWidth>
```
Done.

streamich commented on examples/bfetch_explorer/public/components/double_integers/index.tsx:
```diff
@@ -0,0 +1,109 @@
+  error?: any;
+}
+
+const defaultNumbers = `2000
```
Done.

streamich commented on src/plugins/bfetch/common/buffer/item_buffer.ts:
```diff
@@ -0,0 +1,81 @@
+   * argument which is a list of all buffered items. If `.flush()` is called
+   * when buffer is empty, `.onflush` is called with empty array.
+   */
+  onflush: (items: Item[]) => void;
```
Done.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/)
* Commit: 0bae4eb18f48ad3a914b0f28c961caf2829e1a82

### History
* :broken_heart: [Build #19154](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/) failed eaf01bbfa0626db1bc546c83c2b33f35faeb2f95
* :broken_heart: [Build #19152](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/) failed 2f77faafa1f3571c90b0f08572bc8289e93e708d
* :green_heart: [Build #18928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/) succeeded 148c9362d2f5fa8c2a3da4e564972e9c033bf5ab
* :broken_heart: [Build #18676](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/) failed 627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8
* :green_heart: [Build #18356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/) succeeded 48af298f9c858d5838c8fbf24ef6b8c88216e28a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18356/","number":"18356","commit":"48af298f9c858d5838c8fbf24ef6b8c88216e28a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/","number":"18676","commit":"627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/","number":"18928","commit":"148c9362d2f5fa8c2a3da4e564972e9c033bf5ab"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/","number":"19152","commit":"2f77faafa1f3571c90b0f08572bc8289e93e708d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/","number":"19154","commit":"eaf01bbfa0626db1bc546c83c2b33f35faeb2f95"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/","number":"19235","commit":"0bae4eb18f48ad3a914b0f28c961caf2829e1a82"}]}
PIPELINE-->

streamich commented:
> Question about `fetchStreaming`: Why does it return both an observable and a promise? Couldn't we just return the observable and any consumers can call `toPromise` on it if they want the promise behavior?

Yes, it does not have to return a `promise`, I removed it. It was for legacy reasons, because prior `ajax_stream` would return a `promise` and collect individual messages through a callback, now that the callback is replaced by an observable we don't need the `promise`.

> Regarding `TimedItemBuffer` and `ItemBuffer`, is there any reason these classes need to be separate or can we have only a `TimedItemBuffer` that implements all of the behavior? As far as I can tell there's no place that one is used independently (but I could be wrong).

They are separate because they provide different functionality, it could be combined into one class, but since its already written separately, would like to leave it as is.

> Also, I know you had mentioned adding in functional tests from the example plugin, is that something you're wanting to tackle in this PR or in a future one?

Yes, I added functional tests.

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/)
* Commit: 3ac588d206e62574bc71207cacc9ea3ef939c480

### History
* :broken_heart: [Build #19235](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/) failed 0bae4eb18f48ad3a914b0f28c961caf2829e1a82
* :broken_heart: [Build #19154](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/) failed eaf01bbfa0626db1bc546c83c2b33f35faeb2f95
* :broken_heart: [Build #19152](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/) failed 2f77faafa1f3571c90b0f08572bc8289e93e708d
* :green_heart: [Build #18928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/) succeeded 148c9362d2f5fa8c2a3da4e564972e9c033bf5ab
* :broken_heart: [Build #18676](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/) failed 627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18676/","number":"18676","commit":"627c1eb0ed1fbf6d1b7f036d0caf1541b31144a8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/","number":"18928","commit":"148c9362d2f5fa8c2a3da4e564972e9c033bf5ab"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/","number":"19152","commit":"2f77faafa1f3571c90b0f08572bc8289e93e708d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/","number":"19154","commit":"eaf01bbfa0626db1bc546c83c2b33f35faeb2f95"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/","number":"19235","commit":"0bae4eb18f48ad3a914b0f28c961caf2829e1a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/","number":"19240","commit":"3ac588d206e62574bc71207cacc9ea3ef939c480"}]}
PIPELINE-->

streamich commented:
@elasticmachine merge upstream

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/)
* Commit: 809f56684fee381debf064de6090bc933a80ac46

### History
* :broken_heart: [Build #19240](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/) failed 3ac588d206e62574bc71207cacc9ea3ef939c480
* :broken_heart: [Build #19235](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/) failed 0bae4eb18f48ad3a914b0f28c961caf2829e1a82
* :broken_heart: [Build #19154](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/) failed eaf01bbfa0626db1bc546c83c2b33f35faeb2f95
* :broken_heart: [Build #19152](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/) failed 2f77faafa1f3571c90b0f08572bc8289e93e708d
* :green_heart: [Build #18928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/) succeeded 148c9362d2f5fa8c2a3da4e564972e9c033bf5ab

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/18928/","number":"18928","commit":"148c9362d2f5fa8c2a3da4e564972e9c033bf5ab"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/","number":"19152","commit":"2f77faafa1f3571c90b0f08572bc8289e93e708d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/","number":"19154","commit":"eaf01bbfa0626db1bc546c83c2b33f35faeb2f95"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/","number":"19235","commit":"0bae4eb18f48ad3a914b0f28c961caf2829e1a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/","number":"19240","commit":"3ac588d206e62574bc71207cacc9ea3ef939c480"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/","number":"19315","commit":"809f56684fee381debf064de6090bc933a80ac46"}]}
PIPELINE-->

streamich commented:
@elasticmachine merge upstream

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/)
* Commit: c87291f23db4ccff66da6cc1db8629a090390545

### History
* :broken_heart: [Build #19315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/) failed 809f56684fee381debf064de6090bc933a80ac46
* :broken_heart: [Build #19240](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/) failed 3ac588d206e62574bc71207cacc9ea3ef939c480
* :broken_heart: [Build #19235](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/) failed 0bae4eb18f48ad3a914b0f28c961caf2829e1a82
* :broken_heart: [Build #19154](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/) failed eaf01bbfa0626db1bc546c83c2b33f35faeb2f95
* :broken_heart: [Build #19152](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/) failed 2f77faafa1f3571c90b0f08572bc8289e93e708d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19152/","number":"19152","commit":"2f77faafa1f3571c90b0f08572bc8289e93e708d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/","number":"19154","commit":"eaf01bbfa0626db1bc546c83c2b33f35faeb2f95"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/","number":"19235","commit":"0bae4eb18f48ad3a914b0f28c961caf2829e1a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/","number":"19240","commit":"3ac588d206e62574bc71207cacc9ea3ef939c480"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/","number":"19315","commit":"809f56684fee381debf064de6090bc933a80ac46"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/","number":"19350","commit":"c87291f23db4ccff66da6cc1db8629a090390545"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/)
* Commit: 364826cd70c2e1196e53ca6ef69355a27556f27a

### History
* :broken_heart: [Build #19350](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/) failed c87291f23db4ccff66da6cc1db8629a090390545
* :broken_heart: [Build #19315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/) failed 809f56684fee381debf064de6090bc933a80ac46
* :broken_heart: [Build #19240](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/) failed 3ac588d206e62574bc71207cacc9ea3ef939c480
* :broken_heart: [Build #19235](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/) failed 0bae4eb18f48ad3a914b0f28c961caf2829e1a82
* :broken_heart: [Build #19154](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/) failed eaf01bbfa0626db1bc546c83c2b33f35faeb2f95

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19154/","number":"19154","commit":"eaf01bbfa0626db1bc546c83c2b33f35faeb2f95"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/","number":"19235","commit":"0bae4eb18f48ad3a914b0f28c961caf2829e1a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/","number":"19240","commit":"3ac588d206e62574bc71207cacc9ea3ef939c480"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/","number":"19315","commit":"809f56684fee381debf064de6090bc933a80ac46"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/","number":"19350","commit":"c87291f23db4ccff66da6cc1db8629a090390545"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/","number":"19679","commit":"364826cd70c2e1196e53ca6ef69355a27556f27a"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19711/)
* Commit: 6f76117cdbad5ccb5f3d96c62df9b71d3146ce6e

### History
* :broken_heart: [Build #19679](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/) failed 364826cd70c2e1196e53ca6ef69355a27556f27a
* :broken_heart: [Build #19350](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/) failed c87291f23db4ccff66da6cc1db8629a090390545
* :broken_heart: [Build #19315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/) failed 809f56684fee381debf064de6090bc933a80ac46
* :broken_heart: [Build #19240](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/) failed 3ac588d206e62574bc71207cacc9ea3ef939c480
* :broken_heart: [Build #19235](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/) failed 0bae4eb18f48ad3a914b0f28c961caf2829e1a82

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19235/","number":"19235","commit":"0bae4eb18f48ad3a914b0f28c961caf2829e1a82"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/","number":"19240","commit":"3ac588d206e62574bc71207cacc9ea3ef939c480"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/","number":"19315","commit":"809f56684fee381debf064de6090bc933a80ac46"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/","number":"19350","commit":"c87291f23db4ccff66da6cc1db8629a090390545"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/","number":"19679","commit":"364826cd70c2e1196e53ca6ef69355a27556f27a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19711/","number":"19711","commit":"6f76117cdbad5ccb5f3d96c62df9b71d3146ce6e"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20082/)
* Commit: 62cd03aab3b3e4a9f37323c86367389246d08eac

### History
* :broken_heart: [Build #19711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19711/) failed 6f76117cdbad5ccb5f3d96c62df9b71d3146ce6e
* :broken_heart: [Build #19679](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/) failed 364826cd70c2e1196e53ca6ef69355a27556f27a
* :broken_heart: [Build #19350](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/) failed c87291f23db4ccff66da6cc1db8629a090390545
* :broken_heart: [Build #19315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/) failed 809f56684fee381debf064de6090bc933a80ac46
* :broken_heart: [Build #19240](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/) failed 3ac588d206e62574bc71207cacc9ea3ef939c480

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19240/","number":"19240","commit":"3ac588d206e62574bc71207cacc9ea3ef939c480"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/","number":"19315","commit":"809f56684fee381debf064de6090bc933a80ac46"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/","number":"19350","commit":"c87291f23db4ccff66da6cc1db8629a090390545"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/","number":"19679","commit":"364826cd70c2e1196e53ca6ef69355a27556f27a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19711/","number":"19711","commit":"6f76117cdbad5ccb5f3d96c62df9b71d3146ce6e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20082/","number":"20082","commit":"62cd03aab3b3e4a9f37323c86367389246d08eac"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20804/)
* Commit: bd75508259d2686b246b39dfd3516d35eb6fbf88

### History
* :green_heart: [Build #20082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20082/) succeeded 62cd03aab3b3e4a9f37323c86367389246d08eac
* :broken_heart: [Build #19711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19711/) failed 6f76117cdbad5ccb5f3d96c62df9b71d3146ce6e
* :broken_heart: [Build #19679](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/) failed 364826cd70c2e1196e53ca6ef69355a27556f27a
* :broken_heart: [Build #19350](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/) failed c87291f23db4ccff66da6cc1db8629a090390545
* :broken_heart: [Build #19315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/) failed 809f56684fee381debf064de6090bc933a80ac46

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19315/","number":"19315","commit":"809f56684fee381debf064de6090bc933a80ac46"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/","number":"19350","commit":"c87291f23db4ccff66da6cc1db8629a090390545"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/","number":"19679","commit":"364826cd70c2e1196e53ca6ef69355a27556f27a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19711/","number":"19711","commit":"6f76117cdbad5ccb5f3d96c62df9b71d3146ce6e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20082/","number":"20082","commit":"62cd03aab3b3e4a9f37323c86367389246d08eac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20804/","number":"20804","commit":"bd75508259d2686b246b39dfd3516d35eb6fbf88"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/21135/)
* Commit: bd75508259d2686b246b39dfd3516d35eb6fbf88

### History
* :green_heart: [Build #20804](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20804/) succeeded bd75508259d2686b246b39dfd3516d35eb6fbf88
* :green_heart: [Build #20082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20082/) succeeded 62cd03aab3b3e4a9f37323c86367389246d08eac
* :broken_heart: [Build #19711](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19711/) failed 6f76117cdbad5ccb5f3d96c62df9b71d3146ce6e
* :broken_heart: [Build #19679](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/) failed 364826cd70c2e1196e53ca6ef69355a27556f27a
* :broken_heart: [Build #19350](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/) failed c87291f23db4ccff66da6cc1db8629a090390545

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19350/","number":"19350","commit":"c87291f23db4ccff66da6cc1db8629a090390545"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19679/","number":"19679","commit":"364826cd70c2e1196e53ca6ef69355a27556f27a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/19711/","number":"19711","commit":"6f76117cdbad5ccb5f3d96c62df9b71d3146ce6e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20082/","number":"20082","commit":"62cd03aab3b3e4a9f37323c86367389246d08eac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20804/","number":"20804","commit":"bd75508259d2686b246b39dfd3516d35eb6fbf88"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/21135/","number":"21135","commit":"bd75508259d2686b246b39dfd3516d35eb6fbf88"}]}
PIPELINE-->
